File: Quizify-FrontEnd/src/app/quizify/tsclasses/game.ts
import { Category } from './category';
import { Question } from './question';
import { Admin } from './admin';
import { Topic } from './topic';

export class Game {
     name: string;
     category: Category;
     topic: Topic;
     playcount: number;
     numOfQuestion: number;
     questions: Array<Question>;
     admin: Admin;
     level: string;
}

File: search-service/src/main/java/com/stackroute/quizify/kafka/domain/Game.java
package com.stackroute.quizify.kafka.domain;

import lombok.Data;
import org.springframework.data.annotation.Id;

import java.util.List;

@Data
public class Game {

    @Id
    private long id;
    private String name;
    private Category category;
    private String tag;
    private Topic topic;
    private String genre;
    private String level;
    private int numOfQuestion;
    private List<Question> questions;
    private int liked;
    private int playCount;
    private String rules;
    private String timestamp;
    private Admin admin;
}
